File: src/main/java/burrows/apps/example/gif/presentation/di/module/RiffsyModule.java
package burrows.apps.example.gif.presentation.di.module;

import burrows.apps.example.gif.data.rest.repository.RiffsyApiClient;
import burrows.apps.example.gif.presentation.di.scope.PerActivity;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class RiffsyModule {
  @Provides @PerActivity protected RiffsyApiClient provideRiffsyApi(Retrofit.Builder builder) {
    return builder.baseUrl(RiffsyApiClient.BASE_URL).build().create(RiffsyApiClient.class);
  }
}

File: src/main/java/burrows/apps/example/gif/presentation/di/module/RiffsyModule.kt
package burrows.apps.example.gif.presentation.di.module

import burrows.apps.example.gif.BuildConfig
import burrows.apps.example.gif.data.repository.RiffsyApiClient
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
@Module
class RiffsyModule {
  @Provides fun providesRiffsyApi(retrofit: Retrofit.Builder): RiffsyApiClient = retrofit
    .baseUrl(BuildConfig.BASE_URL)
    .build()
    .create(RiffsyApiClient::class.java)
}
